adsdata/reader.py

```diff
@@ -102,8 +121,9 @@ def read(self, size=-1):
     
 
     def readline(self):
-        line = self._iostream.readline()
-        return self.process_line(line)
+        return self.read()
```

Comment:
this has changed in the latest PR, although the underlying concern may still be valid